src-local/ca_continuation_utils.py

```
@@ -0,0 +1,647 @@
+        if Ca_values:
+            Ca_values[-1] = Ca_critical_refined
+        Ca_critical = Ca_critical_refined
+    elif Ca_fail is None:
+        print(f"\nNo failure found up to Ca = {Ca_coarse[-1]:.{get_decimal_places_from_tolerance(tolerance)}f}")
+    
+    timing_info['stage2_time'] = time.time() - stage2_start
+    
+    # Final extrapolation if we have enough data points near critical Ca
+    if len(Ca_values) >= 3 and len(theta_min_values) >= 3:
+        # Check if we're close to critical point but haven't reached θ_min = 0
+        if min(theta_min_values) > 0.01:  # θ_min still > ~0.5°
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)